### PR TITLE
Static vision model

### LINKED ISSUE #1364: [FEAT] support a df of predictions instead of providing a model
**Is your feature request related to a problem? Please describe.**
Currently, our checks and suites require a compatible model to work with. The 2 main functions that are used are the `predict` function that returns the predictions, and the inferring of `feature_importance`. It limits the package to work with a limited number of models and frameworks.

**Describe the solution you'd like**
Instead of specifying a model, let the user pass a data frame full of predictions (and probably another parameter that specifies the feature_importances). This way we will be model and framework agnostic.

### PR DESCRIPTION
**Pull request by JKL98ISR**
closes #1364
also fixed batch length bug

### CODE REVIEW

File: deepchecks/vision/batch_wrapper.py [hunk 1]
```diff
@@ -49,15 +50,28 @@ def labels(self):
...
     def predictions(self):
         """Return predictions for the batch, formatted in deepchecks format."""
         if self._predictions is None:
             dataset = self._context.get_data_by_kind(self._dataset_kind)
-            # Calling model will raise error if model was not given
-            model = self._context.model
-            self._context.assert_predictions_valid(self._dataset_kind)
-            self._predictions = dataset.infer_on_batch(self._batch, model, self._context.device)
+            if self._context.static_predictions is not None:
+                self._context.assert_predictions_valid(self._dataset_kind)
```
Comment (matanper) [inline deepchecks/vision/batch_wrapper.py:68]: why do you have the same `self._context.assert_predictions_valid(self._dataset_kind)` twice?

File: deepchecks/vision/batch_wrapper.py [hunk 2]
```diff
@@ -75,13 +89,19 @@ def __getitem__(self, index: int):
 
     def __len__(self):
         """Return length of batch."""
-        return len(self._batch)
+        dataset = self._context.get_data_by_kind(self._dataset_kind)
```
Comment (matanper) [inline deepchecks/vision/batch_wrapper.py:92]: I'm not sure this is the best way to get batch length, maybe we should use `len(batch_to_images(batch))` 
@noamzbr WDYT?

**Review by matanper — APPROVED** [review]

File: deepchecks/vision/batch_wrapper.py (hunk 1 above)
Comment (JKL98ISR) [reply deepchecks/vision/batch_wrapper.py:68]: fixed comment bellow

File: deepchecks/vision/batch_wrapper.py (hunk 2 above)
Comment (JKL98ISR) [reply deepchecks/vision/batch_wrapper.py:93]: batch_to_images not always implemented.
fixed case of sampled data

**Review by ItayGabbay — APPROVED** [review]

File: deepchecks/vision/batch_wrapper.py [hunk 3]
```diff
@@ -75,13 +90,19 @@ def __getitem__(self, index: int):
...
```
Comment (noamzbr) [inline deepchecks/vision/batch_wrapper.py:93]: @JKL98ISR Can you explain the logic of this code? Possible also in comments. 
Seems to me like this returns the number of batches in the dataset.

**Pull request merged**